rollout.py

```diff
@@ -35,13 +41,26 @@ def save(path, data):
 def add(info, version, start, duration, barrier=None, deadend=None):
     '''Append a new rollout.  Start is an arbitrary human-readable string;
     duration is in hours.'''
+
+    # Parse and validate date
     import dateparser  # dnf install python3-dateparser
     start_time = dateparser.parse(start, settings={
         'PREFER_DATES_FROM': 'future'
     })
     if start_time is None:
         raise Exception(f"Couldn't parse '{start}'")
 
+    # Validate version
+    if info['stream'] not in VERSION_STREAM_CODES:
+        raise Exception(f"Unknown stream '{info['stream']}'")
+    try:
+        stream_code = int(version.split('.')[2])
```

Comment:
While we're here, we might as well be even stricter. In versionary (the script that generates version strings), we sanity-check what we generated using https://github.com/coreos/fedora-coreos-releng-automation/blob/ff24355d21472a281c181d0c9e952871ce2659d8/scripts/versionary.py#L152-L161. Let's do something similar here?

But cool to do that in a follow-up.